### PR TITLE
ckeditor5: Improved types for PluginCollection.get

### PR DESCRIPTION
**Pull request by fdintino**
This change makes it so that `PluginCollection.get` (i.e. `Editor.plugins.get`) has as its return type the specific `Plugin` subclass when called with a string plugin name.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [`PluginCollection.get()`](https://ckeditor.com/docs/ckeditor5/latest/api/module_core_plugincollection-PluginCollection.html#function-get)
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header (**N/A**)

### CODE REVIEW

File: types/ckeditor__ckeditor5-core/src/plugincollection.d.ts
Comment (fedemp) [inline types/ckeditor__ckeditor5-core/src/plugincollection.d.ts:16]: For the record, this is the most "controversial" line. I'm fine with it, but it would be great if some DT mantainer can approve it too.

File: types/ckeditor__ckeditor5-core/v27/src/plugin.d.ts
Comment (fedemp) [inline types/ckeditor__ckeditor5-core/v27/src/plugin.d.ts:59]: Good catch.

File: types/ckeditor__ckeditor5-easy-image/ckeditor__ckeditor5-easy-image-tests.ts
Comment (fedemp) [inline types/ckeditor__ckeditor5-easy-image/ckeditor__ckeditor5-easy-image-tests.ts:3]: Any reason to remove these tests? Have you moved them somewhere else?

**Review by fedemp — CHANGES_REQUESTED** [review]
Just need to restore tests. I'll approve then.

File: types/ckeditor__ckeditor5-easy-image/ckeditor__ckeditor5-easy-image-tests.ts
Comment (fdintino) [reply types/ckeditor__ckeditor5-easy-image/ckeditor__ckeditor5-easy-image-tests.ts:3]: These tests appear to have been copied from the export to pdf plugin (where they still reside)...I removed them from here because these are the easy-image tests

**Review by fedemp — APPROVED** [review]

**Review by fedemp — APPROVED** [review]

**Comment by fdintino** [comment]
@fedemp who do you usually reach out to for PRs that require DT maintainer approval?

**Review by fedemp — APPROVED** [review]

**Comment by fdintino** [comment]
It looks like this PR requires approval from a DT maintainer. @sandersn @orta , you've both recently reviewed ckeditor5 types PRs. Could you take a look at this? Thanks!

**Comment by fedemp** [comment]
@peterblazejewicz has helped a lot in this front too. :rocket: 

Given the amount of files edited, it would be great to merge this before I try to merge something else.

**Review by sandersn — APPROVED** [review]
Reviewed the OTHER_FILES.txt change. :+1: for reducing the files listed there.

**Review by peterblazejewicz — COMMENTED** [review]
As far I as I can see, there is no exported global symbols from CKEditor by default
It's up to current app to do this, in some way (so there is no `window.CKEDITOR` by default).
So using global symbol declarations is a kind of hack here, allowing to span plugin name resolution across types (to somehow mimic plugins registrations).

What about switching to proper module augmentation, when key is added via agumentated module?
So in '@ckeditor/ckeditor-core` define in plugin collection:

`types/ckeditor__ckeditor-core/src/plugincollection.d.ts`:
```ts

export interface  CKEditorsPlugins {}
...
export default class PluginCollection implements Emitter, Iterable<[typeof Plugin, Plugin]> {
...
    get<T extends keyof CKEditorsPlugins>(key: T): CKEditorsPlugins[T];
...
```

and in e.g. other module which has to import core anyway:

```ts
// types/ckeditor__ckeditor-alignment/index.d.ts
declare module '@ckeditor/ckeditor5-core/src/plugincollection' {
    interface CKEditorsPlugins {
        Alignment: Alignment;
    }
}
```

this should make this work:


```ts
// $ExpectType Alignment
editor.plugins.get('Alignment');
```

that is.. I can see what you're trying to achieve across ESM modules, but we should avoid using global symbols and hacks, to avoid 'webpack's drama' (a lot of creative solutions that with time derived a ton from actual code and make everything harder)

**Review by peterblazejewicz — CHANGES_REQUESTED** [review]
Please review comment about  using augmentation, instead of global symbol

**Comment by fdintino** [comment]
I will update the PR to use module augmentation and re-request review

**Review by peterblazejewicz — APPROVED** [review]
LGTM!

**Comment by fdintino** [comment]
Ready to merge

**Pull request merged**
